src/smart_review/ai/base.py

```diff
@@ -0,0 +1,208 @@
+import logging
+import time
+from abc import ABC, abstractmethod
+from datetime import datetime
+from enum import Enum
+from typing import Any, Dict, List, Tuple
+
+from attrs import define, field
+from github.PullRequestReview import PullRequestReview
+
+from smart_review.exceptions import SmartReviewGithubException
+from smart_review.gitops.github import GitHubClient
+
+logger = logging.getLogger(__name__)
+
+
+class ResponseTypeEnum(str, Enum):
+    """The type of response from the LLM."""
+
+    POSITIVE_REVIEW = "positive_review"
```

Comment:
## Smart Review

Consider adding type hints for better readability and to assist tools like mypy.